### ISSUE TITLE
Bug in WRITEPSF 

### ISSUE THREAD

**Issue by Bellsthescientist** [akohlmey/topotools#26]
Hello, 

I am attempting to merge two pdb files and two psf files using topotools. 

I am using VMD version 1.9.3 and topotools 1.7. 

The problem arises in the psf file generated by topotools. Instead of reading and re-printing the atom type column from the two input psf files, the output from topotools (merged.psf) writes the atom name column again in this atom type column. 

For example: 
Take this one row from my input PSF file: 
1 M0   1    BGLC C1   CC3162   0.290000       12.0110           0

The merged.psf file will write this as follows: 

1 M0   1    BGLC C1   C1     0.290000       12.0110           0

As a consequence, running molecular dynamics is not possible since the parameters are not found for C1

Thank you for your consideration
Bella

**Comment by akohlmey** [comment]
@Bellsthescientist Please try with a more recent version of VMD and TopoTools to make certain the bug has not been fixed yet.  VMD 1.9.3 is almost 8 years old. You can download a more recent version as "alpha" version, even though that is a couple years old as well last time I checked.

**Comment by Bellsthescientist** [comment]
> @Bellsthescientist Please try with a more recent version of VMD and TopoTools to make certain the bug has not been fixed yet. VMD 1.9.3 is almost 8 years old. You can download a more recent version as "alpha" version, even though that is a couple years old as well last time I checked.

Dear Dr. Kohlmeyer, 

Apologies for not doing this due diligence. 

The issue persists with VMD 1.9.4 latest alpha using topotools 1.8 and topotools 1.9 ( I tried both). 

Thank you 
Bella

**Comment by akohlmey** [comment]
> The issue persists with VMD 1.9.4 latest alpha using topotools 1.8 and topotools 1.9 ( I tried both).

Thanks for the update, I will take a look. I am very busy right now with work related to the next stable release of the LAMMPS MD software, so please have a little patience.

**Comment by akohlmey** [comment]
@Bellsthescientist I cannot reproduce the bug.

Please see the attached archive with a minimal demo. As you can see from the "merged.psf" file, atom types are carried over as expected.  Could it be that you are loading the .psf and .pdb files in the wrong order (.pdb first)?

[merge_demo.tar.gz](https://github.com/user-attachments/files/16521377/merge_demo.tar.gz)

**Comment by Bellsthescientist** [comment]
Dear Dr. Kohlmeyer, 

Thank you for providing this. I tried your .vmd and it worked perfectly as you also found. Here is what I was doing previously that was wrong somehow: 

I first loaded the two molecule pdbs and their individual psf files as two separate IDs {0} and {1} in VMD and then I ran these commands in the tk console: This would result in the incorrect PSF file as I mentioned before. 

package require topotools
>>> namespace import TopoTools::mergemols
>>>
>>> set newmol [mergemols {0 1}]
>>>
>>> animate write psf merge.psf $newmol
>>> animate write pdb merge.pdb $newmol
